tools/loader/ImageClassifier.cpp

```diff
@@ -83,11 +83,18 @@ llvm::cl::opt<ImageNormalizationMode> imageMode(
 llvm::cl::alias imageModeA("i", llvm::cl::desc("Alias for -image_mode"),
                            llvm::cl::aliasopt(imageMode),
                            llvm::cl::cat(imageLoaderCat));
+
+
+llvm::cl::opt<std::string> loadimageFormat(
```

Comment:
Instead of having this as a string, can you create an enum for it?

Reply:
Hi Jordon, 

I made the following changes - 
enum class ImageLayout {
  /// N x C x H x W 
  NCHW,
  /// N x H x W x C
  NHWC,
};
ImageLayout strToImageLayout(const std::string &str) {
  return llvm::StringSwitch<ImageLayout>(str)
      .Case("nhwc", ImageLayout::NHWC)
      .Case("nchw", ImageLayout::NCHW);
  GLOW_ASSERT(false && "Unknown image layout");
}
llvm::cl::opt<ImageLayout> imageLayout(
    "image_layout", llvm::cl::desc("Specify which image layout to use"), 
    llvm::cl::Optional,
    llvm::cl::values(
        clEnumValN(ImageLayout::NCHW, "nchw",
                   "Use NCHW image layout"),
        clEnumValN(ImageLayout::NHWC, "nhwc",
                   "Use NHWC image layout")),
    llvm::cl::init(ImageLayout::NCHW));

I can see that it is being passed on the command line correctly, I am using the CLI passed in this function - 
void loadImagesAndPreprocess(const llvm::cl::list<std::string> &filenames,
                             Tensor *result, ImageNormalizationMode normMode, ImageLayout layout) 

switch (layout) {
  case ImageLayout::NCHW:
    result->reset(ElemKind::FloatTy,
                {numImages, numChannels, imgHeight, imgWidth});
  case ImageLayout::NHWC:
    result->reset(ElemKind::FloatTy,
                {numImages, imgHeight, imgWidth, numChannels});
  default:
    GLOW_ASSERT(false && "Invalid Image Layout");
  }

I can see that the command line parameter takes in 0 for nchw and 1 for nhwc, when I print it inside this function. However, it is not being matched correctly in the switch/case (although the enum also takes int values). I am puzzled that I am missing some link here. I referred to the imageMode code and I cannot figure where the strToImageNormalizationMode function is being called or if it is needed.

Reply:
changed formatting, you can use "```" and cpp to format correctly.
in your switch you are missing breaks. Not sure if it's the only problem though.

Reply:
it worked, thanks :)